src/distribution.cpp

```diff
@@ -19,32 +19,31 @@
 namespace openmc {
 
 //==============================================================================
-// Discrete implementation
+// DiscreteIndex implementation
 //==============================================================================
 
-Discrete::Discrete(pugi::xml_node node)
+DiscreteIndex::DiscreteIndex(pugi::xml_node node)
 {
   auto params = get_node_array<double>(node, "parameters");
+  std::size_t n = params.size() / 2;
 
-  std::size_t n = params.size();
-  std::vector<double> x_vec(params.begin(), params.begin() + n / 2);
-  std::vector<double> p_vec(params.begin() + n / 2, params.end());
+  assign(&(*params.begin()) + n, n);
```

Comment:
I think this should work and is slightly prettier
```suggestion
  assign(params.data() + n, n);
```

Reply:
Thanks for your patience with my rusty coding...